app/src/parser_print_helper.c

```diff
@@ -189,28 +189,27 @@ parser_error_t page_principal_with_subaccount(const uint8_t *sender, uint16_t se
     // crc is computed with full subaccount with all zeros at the beginning
     MEMCPY(tmpArray + senderLen, fromSubaccount, fromSubaccountLen);
     uint32_t crc = 0;
-    crc32_small(tmpArray, DFINITY_PRINCIPAL_LEN + DFINITY_SUBACCOUNT_LEN, &crc);
+    crc32_small(tmpArray, senderLen + fromSubaccountLen, &crc);
     crc_array[0] = (uint8_t) ((crc & 0xFF000000) >> 24);
     crc_array[1] = (uint8_t) ((crc & 0x00FF0000) >> 16);
     crc_array[2] = (uint8_t) ((crc & 0x0000FF00) >> 8);
     crc_array[3] = (uint8_t) ((crc & 0x000000FF) >> 0);
-    // for some reason this function returns -1 if there's an error
     uint8_t crcLen = (uint8_t) base32_encode(crc_array, 4, crc_text, sizeof(crc_text));
     if (crcLen == 0) {
         return parser_unexpected_error;
     }
 
     // print checksum
     MEMCPY(text_ptr, crc_text, crcLen);
-    snprintf(text_ptr + crcLen, 2, "%c", SEPARATOR);
+    *(text_ptr + crcLen) = SEPARATOR;
     text_ptr += crcLen + 1;
 #if !defined(TARGET_STAX) // needed if crc32 length is < 7
     for (uint8_t i = crcLen; i < 7; i++) {
-        snprintf(text_ptr, 2, " ");
-        text_ptr += crcLen + 1;
+        *text_ptr = ' ';
+        text_ptr++;
```

Comment:
Is this correct? Before it was `text_ptr += crcLen + 1;` but now it's just `text_ptr++;`

Reply:
It was wrong before.
As it was overwritten later in the file, it didn't raise a bug.